### PR TITLE
feat: GET payment by id

### PR DESCRIPTION
**Pull request by NjengaC**
## Description
This pull request implements  the  GET payment by id endpoints in the FastAPI application:

- [x] GET /payment/{payment_id}}: Retrieves all payment details for the specific id provided. This endpoint returns a payment object with details for that specific payment.

[[## Related Issue (Link to issue https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/454

### Motivation and Context
These changes were necessary to implement functionality for retrieving payment details for users either as a complete list or filtered by a specific user. Proper response handling and validation will be added to ensure robustness and consistency of the API.

​

### How Has This Been Tested?

- GET /payment/{payment_id}: Tested by retrieving payment details for specific users and ensuring correct user-based filtering.

Testing was conducted using local server requests, pytests and cURL commands. All responses were validated against the expected models.
​

      ​
Screenshots
image

### Changes Made:

1. Added  PaymentResponse and PaymentService schema and service to handle request and response formats.

2. Implemented GET /payment/payment_id to return payment details for a specific user using the Payment object model.

### Types of changes
 

- [ ] Bug fix (non-breaking change which fixes an issue)

 

- [x] New feature (non-breaking change which adds functionality)

 

- [ ] Breaking change (fix or feature that would cause existing functionality to change)

###Checklist:

- [x]  My code follows the code style of this project.

- [x]  My change requires a change to the documentation.

 

- [x] I have updated the documentation accordingly.

- [x]  I have read the CONTRIBUTING document.

 

- [x] I have added tests to cover my changes.

 

- [x] All new and existing tests passed.

### CODE REVIEW

File: test.db
Comment (joboy-dev) [inline test.db:1]: Remove this from git

File: api/v1/services/payment.py
Comment (joboy-dev) [inline api/v1/services/payment.py:1]: No check to see if the payment exists??

There's a function called check_model_existence. Use that and you solve all your problems

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes

File: api/v1/services/payment.py
Comment (NjengaC) [reply api/v1/services/payment.py:1]: Noted, Incorporated the method.

File: api/v1/routes/payment.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from api.utils.dependencies import get_db
+from api.v1.schemas.payment import PaymentResponse
+from api.utils.db_validators import check_model_existence
+
+payment = APIRouter()
+
+@payment.get("/payments/{payment_id}", response_model=PaymentResponse)
+async def get_payment(payment_id: str, db: Session = Depends(get_db)):
+    payment_service = PaymentService()
+    payment = check_model_existence(db, payment_service.model, payment_id)
```
Comment (joboy-dev) [inline api/v1/routes/payment.py:13]: This should be in services and should be like this
`check_model_existence(db, Payment, payment_id)` where Payment is the payment database model

File: api/v1/services/payment.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from sqlalchemy.orm import Session
+from api.v1.models.payment import Payment
+
+class PaymentService:
+    model = Payment
+
+    def get_payment_by_id(self, db: Session, payment_id: str):
+        return db.query(Payment).filter(Payment.id == payment_id).first()
```
Comment (joboy-dev) [inline api/v1/services/payment.py:8]: Change this based on the first comment

**Review by joboy-dev — CHANGES_REQUESTED** [review]
make changes

File: api/v1/routes/payment.py (hunk 1 above)
Comment (NjengaC) [reply api/v1/routes/payment.py]: I have refactored to include check_model_existence in services and call it in get_payment_by_id inside routes

File: api/v1/services/payment.py (hunk 2 above)
Comment (NjengaC) [reply api/v1/services/payment.py]: Done

**Review by joboy-dev — APPROVED** [review]

**Review by SundayMba — APPROVED** [review]

**Pull request merged**